src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsTag.cs

```diff
@@ -0,0 +1,167 @@
+            block.Arrange(new Rect(block.DesiredSize));
+
+            var lineHeight = Math.Floor(format.Typeface.FontFamily.LineSpacing * block.FontSize);
+            var image = new CrispImage
+            {
+                Moniker = GetMoniker(),
+                MaxHeight = lineHeight,
+                Margin = new Thickness(1, 0, 5, 0)
+            };
+
+            var stackPanel = new StackPanel
+            {
+                Height = lineHeight,
+                Orientation = Orientation.Horizontal
+            };
+
+            stackPanel.Children.Add(image);
+            stackPanel.Children.Add(block);
+
+            var border = new Border
+            {
+                BorderBrush = format.BackgroundBrush,
+                BorderThickness = new Thickness(1),
+                Background = Brushes.Transparent,
+                Child = stackPanel,
+                CornerRadius = new CornerRadius(2),
+                // Highlighting lines are 2px buffer. So shift us up by one from the bottom so we feel centered between them.
+                Margin = new Thickness(10, top: 0, 0, bottom: 1),
+                Padding = new Thickness(1)
+            };
+
+            border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            // Need to set these properties to avoid unnecessary reformatting because some dependancy properties
+            // affect layout.
+            // TODO: Not sure if these are needed anymore since the errors are not intratextadornment tags
+            TextOptions.SetTextFormattingMode(border, TextOptions.GetTextFormattingMode(view.VisualElement));
+            TextOptions.SetTextHintingMode(border, TextOptions.GetTextHintingMode(view.VisualElement));
+            TextOptions.SetTextRenderingMode(border, TextOptions.GetTextRenderingMode(view.VisualElement));
+
+            view.ViewportWidthChanged += ViewportWidthChangedHandler;
+
+            return new GraphicsResult(border, dispose:
+                () =>
+                {
+                    link.RequestNavigate -= HandleRequestNavigate;
+                    view.ViewportWidthChanged -= ViewportWidthChangedHandler;
+                });
+
+            void ViewportWidthChangedHandler(object s, EventArgs e)
+            {
+                if (Location is InlineDiagnosticsLocations.PlacedAtEndOfEditor)
+                {
+                    Canvas.SetLeft(border, view.ViewportWidth - border.DesiredSize.Width);
+                }
+            }
+        }
+
+        private ImageMoniker GetMoniker()
+        {
+            switch (_diagnostic.Severity)
+            {
+                case DiagnosticSeverity.Warning:
+                    return KnownMonikers.StatusWarning;
+                case DiagnosticSeverity.Error:
+                    return KnownMonikers.StatusError;
+                default:
+                    throw new MissingMethodException();
```

Comment:
wat

Reply:
lol this was just in while I need to figure out what to do for edit and continue, I either need to find a good Moniker to use or not show anything for that case